### ISSUE TITLE
[SF][Feature Request]: Support passing domRef to ToolbarButton

### ISSUE THREAD

**Issue by ksblklu02** [SAP/ui5-webcomponents#12029]
### Feature Request Description

For a11y requirement, when we open a dialog/popover by clicking a ToolbarButton, the focus should be returned to the ToolbarButton once the dialog/popover is closed. So we need to manually set focus to the ToolbarButton by using React Ref.

### Proposed Solution

Enable passing dom ref to the Toolbar button to manually set focus on it

### Organization

SAP SuccessFactors

### Priority

Medium

### Privacy Policy

- [x] I’m not disclosing any internal or sensitive information.

**Comment by kgogov** [comment]
Hello @ksblklu02,

Thank you for reporting this incident!

We recently made changes to `ui5-toolbar` items in the following pull request:
🔗 https://github.com/SAP/ui5-webcomponents/pull/11374
The described behavior should now be working as expected.

I’ve created a simple example demonstrating how the focus correctly returns to the ToolbarButton after the dialog is closed — even without using a React ref:
🔗 https://stackblitz.com/edit/github-hlyj8c-ppxairtm?file=src%2FApp.tsx

Could you please verify this on your end?

Thank you!

Best regards,
Konstantin
